runtime.go

```diff
@@ -1528,6 +1533,7 @@ func (r *Runtime) RunProgram(p *Program) (result Value, err error) {
 		vm.clearStack()
 	} else {
 		vm.stack = nil
+		vm.profileTicks()
```

Comment:
Without this we would miss the ticks tracked for the above vm.runTry

```
	ex := vm.runTry(r.vm.ctx)
	if ex == nil {
		result = r.vm.result
	} else {
		err = ex
	}
	if recursive {
		vm.popCtx()
		vm.halt = false
		vm.clearStack()
	} else {
		vm.stack = nil
		vm.profileTicks()
		vm.prg = nil
		vm.setFuncName("")
		r.leave()
	}
```

It could turn out that this isn't necessary because it is captured by the pushCtx/restoreCtx, but I don't think it hurts to have this in place just to make sure we capturing any lingering ticks for the program. Worst case scenario is there are no new ticks and we add 0 for the program.

If we think this clutters the code I can dig more into this and see if it is necessary.

Reply:
Yeah without this we would miss a few ticks in a few edge cases, not the end of the world but probably nice to keep around.

![image](https://github.com/mongodb-forks/goja/assets/6693710/f68906db-983c-40c9-8bbf-051285d511c6)